src/scala/com/github/johnynek/bazel_deps/GradleResolver.scala

```diff
@@ -42,203 +37,214 @@ class GradleResolver(
-        case (g, (n, deps)) =>
-          for {
-            cnode <- toCoord(n)
-            transitive <- cats
-              .Traverse[List]
-              .sequence(deps.transitive.getOrElse(Nil).map(toCoord(_)))
-          } yield {
-            val g1 = g.addAllNodes(cnode :: transitive)
-            transitive.foldLeft(g1) { case (g, revDep) =>
-              val curEdge = (
-                s"${revDep.group.asString}:${revDep.artifact.artifactId}",
-                s"${cnode.group.asString}:${cnode.artifact.artifactId}"
+                  Validated.valid(MavenCoordinate(
+                    MavenGroup(org),
+                    MavenArtifactId(
+                      nme,
+                      gradleTpe.getContentTypeOverride.getOrElse(k, "jar"),
+                      ""
+                    ),
+                    Version(resolvedInfo.locked.getOrElse(""))
+                  ))
+                }
+                else {
+                  Validated.invalidNec((k, s"could not parse key $k into maven artifact"))
+                }
+              case None =>
+                Validated.invalidNec((k, s"Unable to lookup info about $k in dep map"))
+            })
+
+        depMap
+          .toList
+          .sortBy(_._1)
+          .traverse[V, (MavenCoordinate, List[MavenCoordinate])] { case (n, deps) =>
+            toCoord(n)
+              .product(
+                deps.transitive.getOrElse(Nil)
+                  .traverse[V, MavenCoordinate](toCoord(_))
               )
-              if (
-                gradleTpe.ignoreDependencyEdge
-                  .map(_.contains(curEdge))
-                  .getOrElse(false)
-              ) {
-                g
-              } else {
-                g.addEdge(Edge(revDep, cnode, ()))
+          } match {
+          case Validated.Valid(edgeList) =>
+            Success(edgeList.foldLeft(Graph.empty[MavenCoordinate, Unit]) { case (g, (cnode, transitive)) =>
+              transitive.foldLeft(g.addNode(cnode)) { case (g, revDep) =>
+                // addEdge adds both nodes in the edge
+                // but if we ignore the edge, we defensively make sure we add revDep
+                // since this graph is validated, it should be added when it appears
+                // as a key however
+                if (ignoreEdge(revDep, cnode)) g.addNode(revDep)
+                else g.addEdge(Edge(revDep, cnode, ()))
               }
-            }
-          }
-      }
-
-      gradleDependencyGraphTry.map { graph =>
-        graph.addAllNodes(m.dependencies.roots)
+            })
+          case Validated.Invalid(errs) =>
+            Failure(new Exception(errs.map(_._2)
+              .mkString_("failed to find ${errs.length} nodes:\n\t", "\n\t", "")))
+        }
       }
-    }
-  }
 
   // Build the entire transitive graph of a set of coordinates
-  def buildGraph(
-      coords: List[MavenCoordinate],
+  private def buildGraph(
```

Comment:
this was here because of a trait, but we ignored an arg, that's code smell, so I added a new trait (CustomResolver).